### DISCUSSION TITLE
Confirmation dialog

### DISCUSSION THREAD

**Discussion by vytautas-pranskunas-** [taiga-family/taiga-ui#9498] [Q&A] [answered]
Hi,

where can i get normal confiormation dialog code? I found many dialogs on your UI library docs but not a standard dialog with message, yes no buttons. Where no would just close dialog with no action and yes would proceed with an action.

When possible it would be good to have inline template so it would be easy to do resuable confirmation dialog service.

Thanks

**Comment by waterplea** [comment, marked as answer]
It's here:
https://taiga-ui.dev/components/confirm

**Reply by vytautas-pranskunas-** [reply]
uff. Thanks. I really strugling navigatineg in your component library documentation. Not sure if thats my problem or UX.

**Reply by vytautas-pranskunas-** [reply]
btw is there any specific reason why buttons in dialog does not follow buthon theme? or should i apply separate appereance to them. 
Now i have this: tuiButtonOptionsProvider({ appearance: 'parama-button', size: 'm' }), 
_p.s. i do not see appereance option in tuiMobileDialogOptionsProvider_

also thats strange that modal-dialog doe snot use --tui-radius-s or --tui-radius-m variables... 
![Screenshot 2024-10-15 at 11 28 35](https://github.com/user-attachments/assets/079da36b-99bc-453e-9380-c4a2239835ef)

**Comment by waterplea** [comment]
You can pass `appearance` as part of `TuiConfirmData` for the "yes" button, in case, for example, you are confirming a deletion and want it to look destructive/red. But secondary button is predefined to be secondary on mobile and flat on desktop per our design specs. It's fairly easy to recreate such a dialog and reuse your own implementation if you need it to behave differently, take a look at the source code for inspiration.

`tuiMobileDialogOptionsProvider` is for a different component, that was supposed to look like a native dialog on Android/iOS, but I believe it's a bit dated:
https://taiga-ui.dev/components/mobile-dialog

As for variable, recently the spec was updated and border-radius now matches `--tui-radius-xl`, previously it was a bit bigger if I remember correctly. I'm hesitant to use a variable there just yet though, as we might want to revisit radius variables as part of updating the specs, since radiuses are not that well defined right now as to where which is supposed to be used.

**Reply by vytautas-pranskunas-** [reply]
But if i want to change radius, should i change it manually via css or through appereance?

**Reply by waterplea** [reply]
For now you can do it with appearance, yes. You can use `tuiDialogOptionsProvider` to set default appearance and in global styles change radius, padding etc. Just make sure it has high enough specificity if you do not see your changes applied.
